tools/txg/TXGRepl.hs

```diff
@@ -0,0 +1,298 @@
+{-# LANGUAGE CPP                        #-}
+{-# LANGUAGE DeriveAnyClass             #-}
+{-# LANGUAGE DeriveFunctor              #-}
+{-# LANGUAGE DeriveGeneric              #-}
+{-# LANGUAGE DerivingStrategies         #-}
+{-# LANGUAGE FlexibleContexts           #-}
+{-# LANGUAGE LambdaCase                 #-}
+{-# LANGUAGE OverloadedStrings          #-}
+{-# LANGUAGE ScopedTypeVariables        #-}
+{-# LANGUAGE TupleSections              #-}
+{-# language GeneralizedNewtypeDeriving #-}
+
+{-# OPTIONS_GHC -fno-warn-missing-signatures #-}
+
+module TXGRepl where
+
+import Control.Monad
+import Control.Monad.Catch
+import Control.Monad.Reader
+import Control.Monad.State
+import Control.Lens hiding ((.=))
+
+import Data.Aeson
+import Data.Aeson.Types
+-- import Data.ByteString (ByteString)
+import Data.Decimal
+import Data.Foldable
+import qualified Data.List.NonEmpty as NEL
+import Data.Proxy
+import qualified Data.Text as T
+import Data.Text (Text)
+
+import GHC.Generics
+
+import Network.HTTP.Client hiding (Proxy(..))
+import Network.HTTP.Client.TLS
+import Network.X509.SelfSigned
+
+import Servant.API
+import Servant.Client
+
+import System.Random
+
+import Text.Printf
+
+-- pact imports
+
+import Pact.ApiReq
+import Pact.Types.API
+import Pact.Types.ChainMeta
+import qualified Pact.Types.ChainId as Pact
+import Pact.Types.Crypto
+
+-- chainweb imports
+
+import Chainweb.ChainId
+import Chainweb.HostAddress
+import Chainweb.Version
+import Chainweb.Pact.RestAPI
+#if !MIN_VERSION_servant(0,15,0)
+import Chainweb.RestAPI.Utils
+#endif
+
+import TXG.Simulate.Utils
+
+-- for ghci
+
+data Builtin
+    = Coin
+    | Hello
+    | Payments
+    deriving (Show, Eq, Ord, Bounded, Enum, Generic)
+    deriving anyclass (FromJSON, ToJSON)
+
+type Keyset = [SomeKeyPair]
+
+type Guard = Keyset
+
+type Account = String
+
+type SenderName = String
+
+type ReceiverName = String
+
+type Balance = Decimal
+
+type Amount = Decimal
+
+data CoinCode
+    = CoinCreateAccount Account Guard
+    | CoinAccountBalance Account
+    | CoinTransfer SenderName ReceiverName Amount Keyset
```

Comment:
We now have a new transfer function that doesn't require the `Keyset`.  Might be good to support that too.

Reply:
Ok, will do so